mobile_app_connector/models/request.py

```diff
@@ -99,3 +104,29 @@ def create_email_for_interaction_resume(self, subject, body, partner):
                 'date': fields.Datetime.now(),
             }).id
         })
+
+    @api.model
+    def detect_lang(self, text):
+        """
+        Use detectlanguage API to find the language of the given text
+        :param text: text to detect
+        :return: res.lang compassion record if the language is found, or False
+        """
+        detectlanguage.configuration.api_key = config.get(
+            'detect_language_api_key')
+        language_name = False
+        langs = detectlanguage.languages()
+        try:
+            code_lang = detectlanguage.simple_detect(text)
+        except (IndexError, detectlanguage.DetectLanguageError):
+            # Language could not be detected
+            return False
+        for lang in langs:
+            if lang.get("code") == code_lang:
+                language_name = lang.get("name")
+                break
+        if not language_name:
+            return False
+
+        return self.env['res.lang.compassion'].search(
+            [('name', '=ilike', language_name)], limit=1)
```

Comment:
Remove this! It's already defined in crm_request module

Reply:
I thought it was the way to do it as it was already done in 2 modules, I will change it